mod_reforged/hooks/items/weapons/weapon.nut

```diff
@@ -1,5 +1,16 @@
+::Reforged.HooksMod.hookTree("scripts/items/weapons/weapon", function(q) {
+	q.create = @(__original) function()
+	{
+		__original();
+		if (this.m.Reach == 0 && this.isItemType(::Const.Items.ItemType.MeleeWeapon))
+		{
+			this.assignDefaultReach();
+		}
+	}
+});
+
 ::Reforged.HooksMod.hook("scripts/items/weapons/weapon", function(q) {
-	q.m.Reach <- 1;
+	q.m.Reach <- 0;
```

Comment:
0 is a valid reach value. Some mod might want to assign this reach to a melee weapon, but that would now be overwritten by assignDefaultReach

Reply:
I suggest `null` here.
If any mod builds acknowledges Reforgeds existance by adjusting Reach values it should also take the extra step of ordering itself after Reforged

Reply:
I see pros and cons of both. If we go with `null` then we have to also assign `0` reach manually for ranged weapons in our `assignDefaultReach` function. If we go with `0` then the modder who wants 0 reach for a melee weapon could theoretically overwrite `assignDefaultReach` with an empty function in that weapon's file.

I am leaning more towards `null` personally. Or basically I am leaning more towards a non-zero value. So another idea could be to do `-1`. Then if someone sets it to `0` then it's fine and works out of the box. But the advantage of going with `-1` is that no one is thrown off by an exception due to `arithmetic operation between integer and null` if he tries to +/- the default reach. But there are edge cases with that as well where if someone modifies that base value then it may clash with what this function is trying to detect.

I'm happy to go with `null` for now. Let's see if @Enduriel has any other thoughts.

Reply:
I actually prefer darxos idea of like -100. So that you can assume values >=0 are not meant to be changed and values <0 are meant to still be adjusted but this allows you to do things like this.m.Range -= 1

Reply:
Either way works for me. `null` seems a bit better to understand when [re]reading the code. And the negative number makes some reach manipulation by submods a bit easier